### PR TITLE
feat(store): find methods

### LINKED ISSUE #40: Store - find methods
* find(filter)
* findOne(filter)
* findOneOrFetch(filter, SOURCE)

### PR DESCRIPTION
**Pull request by JohannesMerz**
-find
-findOne
-findOneOrFetch

closes #40

### CODE REVIEW

File: src/Store.js [hunk 1]
```diff
@@ -2,42 +2,141 @@ import {
...
+        }
+      }
     }
+    return undefined;
+  }
 
-    this.storage = Storage || new EmptyTransporter();
+  _createAndFetchFrom(values, source) {
+    const item = new this._Item({ store: this, autoSave: this._options.autoSave });
+    item.construct(values, { source });
```
Comment (JohannesMerz) [inline src/Store.js:109]: @tchock 
forgot to push the item to the store
  maybe we should create a second item array where all the items, that aren't in the first array (items that are being deleted when constructing the store, empty placeholder items, that are being fetched, etc), and return them if findOrFetched (maybe even the other finds?) doesn't find them in the main array. This way the main array contains only items worth displaying and the 'hidden' Item array prevents Items from being created twice while their action is in progress

**Review by JohannesMerz — COMMENTED** [review]
not quite ready

**Comment by JohannesMerz** [comment]
@tchock removed merge conflicts and added `store.incompleteItems`, i think this should be ready for review & merge

File: src/Store.js [hunk 2]
```diff
@@ -90,12 +104,25 @@ export default class Store {
...
   _createItems(rawItems, source) {
     return Promise.all(rawItems.map((rawItem) => {
       const item = new this._Item({ store: this, autoSave: this._options.autoSave });
-      if (rawItem._transporterState !== STATE.BEING_DELETED &&
-        rawItem._transporterState !== STATE.DELETED &&
-        rawItem._transporterState !== STATE.LOCKED) {
+      if (rawItem._transporterState === STATE.BEING_DELETED ||
+        rawItem._transporterState === STATE.DELETED ||
+        rawItem._transporterState === STATE.LOCKED) {
+        this.incompleteItems.push(item);
```
Comment (tchock) [inline src/Store.js:124]: Do you want to leave these items in the incompleteItems array? Because I can't see where they are removed from there.

**Review by tchock — APPROVED** [review]

**Pull request merged**